test/integration/header.spec.ts

```diff
@@ -181,7 +181,7 @@ describe("typescript", function() {
 
       it("should send and receive duration type headers", async function() {
         const duration = "P123DT22H14M12.011S";
-        await testClient.header.paramDuration("valid", duration);
+        await testClient.header.paramDuration(duration, "valid");
```

Comment:
General comment but this is a good example. Why did the positions of these parameters change? Is this the position we would have had with autorest.typescript v4/v5? Main reason I ask was because we got some feedback asking that when possible, we keep the positional arguments in the same order to make it easier for customers to upgrade SDKs.

Reply:
Yeah, I don't love this change of positions. This happened because I had to make sure that optional parameters always show up after required ones, however it also affected other parameters.

I'll think on a way to keep these as consistent with the old version as possible. If you have any suggestions I'd love to hear them

Reply:
I've integrated your suggestion below to fix this, thanks!